bionic/utils/reload.py

```diff
@@ -0,0 +1,40 @@
+import builtins
+from importlib import reload
+from sys import modules as module_registry
+
+
+def recursive_reload(module):
+    """
+    Helper method to reload a module recursively. If a module imports a set of
+    modules, then the modules in the set are also reloaded and so on.
+
+    Not all modules can be reloaded. This method attempts to reload as many
+    modules as possible.
+
+    Also note that this method may not be able to handle dynamic imports that
+    only happens at runtime. For example, if a module imports another module
+    only when a certain method is executed, reloading the former module does
+    not guarantee that the latter module is reloaded.
+    """
+
+    original_import = builtins.__import__
+    already_reloaded = set()
+
+    def custom_import(name, globals=None, locals=None, fromlist=[], level=0):
+        if name in module_registry:
+            module = module_registry[name]
+            if name not in already_reloaded and hasattr(module, "__file__"):
+                already_reloaded.add(name)
+                try:
+                    reload(module)
+                except Exception:
+                    # The goal is to reload as many modules as possible. If a
```

Comment:
Why can some modules not be reloaded? What kind of exception will they throw? What about swallowing legitimate exceptions -- how we will we deal with that?

For example, if the user has a bug in their new module that throws an exception, won't this swallow it?

Reply:
That is a very good point.

Reply:
Hmm, the test actually passes without the try catch.

Reply:
I'm curious what errors you originally had in mind when you wrote the try-catch. Are there some modules you don't think will reload properly?

Reply:
I think some earlier iteration of this method may have failed to reload certain internal modules. Removed the try-catch as it is no longer necessary.

Reply:
Also updated the comments in the method.